request_handler.h

```diff
@@ -0,0 +1,72 @@
+// For the Request and Response classes, you need to implement the methods
+// and add private data as appropriate. You may also need to modify or extend
+// the API when implementing the reverse proxy. Use your good judgment.
+
+#ifndef REQUEST_HANDLER_H
+#define REQUEST_HANDLER_H
+
+#include <map>
+#include <memory>
+#include <string>
+#include "config_parser.h"
+#include "request.h"
+#include "response.h"
+
+// Represents the parent of all request handlers. Implementations should expect to
+// be long lived and created at server constrution.
+class RequestHandler {
+ public:
```

Comment:
Indentation and style issues for this file will need to be fixed to be like ours.